system/jlib/jsmartsock.cpp

```diff
@@ -332,7 +329,8 @@ ISmartSocket *CSmartSocketFactory::connect_timeout( unsigned timeoutms)
         StringBuffer s("CSmartSocketFactory::connect ");
         ep.getUrlStr(s);
         EXCLOG(e,s.str());
-        ss->status=false;
+        if (ss)
+            ss->status=false;
         if (sock)
             sock->Release();
         throw;
```

Comment:
Previously, the only other exception it threw was the ISmartSocketException, before/outside the try/catch when ss was null (which is still there)

Now it can throw ISmartSocketException if sock is null, but that could be handled in connect_sock too (is it even possible for a null ISocket to be returned from connect_sock()?)